src/workflows/airqo_etl_utils/data_validator.py

```
@@ -81,40 +115,50 @@ def get_valid_value(column_name: str, row_value: Any) -> Any:
+        validated_columns = list(chain.from_iterable(filtered_columns.values()))
 
-        for col in columns:
-            name = configuration.AIRQO_DATA_COLUMN_NAME_MAPPING.get(col, None)
-            data.loc[:, col] = data[col].apply(
+        for col in validated_columns:
+            is_airqo_network = data["network"] == "airqo"
+            mapped_name = configuration.AIRQO_DATA_COLUMN_NAME_MAPPING.get(col, None)
+            data.loc[is_airqo_network, col] = data.loc[is_airqo_network, col].apply(
                 lambda x: DataValidationUtils.get_valid_value(
-                    column_name=name, row_value=x
+                    column_name=mapped_name, row_value=x
```

Ruff (0.8.0)

Function definition does not bind loop variable `mapped_name`

(B023)